cluster/validation.go

```diff
@@ -731,3 +739,19 @@ func getClusterVersion(version string) (semver.Version, error) {
 	}
 	return parsedVersion, nil
 }
+
+// warnWeaveDeprecation prints a deprecation warning if version is higher than 1.27
+func warnWeaveDeprecation(k8sVersion string) error {
+	version, err := util.StrToSemVer(k8sVersion)
+	if err != nil {
+		return fmt.Errorf("failed to translate: %s to semver notation: %v", k8sVersion, err)
```

Comment:
Nit: Any reason we can't use error wrapping? https://stackoverflow.com/a/61287626
```suggestion
		return fmt.Errorf("failed to translate: %s to semver notation: %w", k8sVersion, err)
```